osbs/build/build_requestv2.py

```diff
@@ -625,27 +544,35 @@ def is_custom_base_image(self):
         Returns whether or not this is a build from a custom base image
         """
         return bool(re.match('^koji/image-build(:.*)?$',
-                             self.base_image or ''))
+                             self.user_params.base_image.value or ''))
 
     def is_from_scratch_image(self):
         """
         Returns whether or not this is a build `FROM scratch`
         """
-        return self.base_image == 'scratch'
+        return self.user_params.base_image.value == 'scratch'
 
 
 class SourceBuildRequest(BaseBuildRequest):
     """Build request for source containers"""
-    def __init__(self, build_json_store, outer_template=None):
+    def __init__(self, osbs_api, outer_template=None, user_params=None):
         """
         :param build_json_store: str, path to directory with JSON build files
         :param outer_template: str, path to outer template JSON
         """
+        if user_params:
+            assert isinstance(user_params, SourceContainerUserParams)
+        else:
+            user_params = SourceContainerUserParams()
         super(SourceBuildRequest, self).__init__(
-            build_json_store,
-            outer_template=outer_template or DEFAULT_SOURCES_OUTER_TEMPLATE
+            osbs_api,
+            outer_template=outer_template or DEFAULT_SOURCES_OUTER_TEMPLATE,
+            user_params=user_params,
         )
-        self.user_params = SourceContainerUserParams(self.build_json_store)
+
+    def set_params(self, user_params):
```

Comment:
do we need this? user params are passed already directly to __init__

Reply:
In all three instances, you can create a BuildRequest without a valid user_params and then call set_params later, which is currently used in some of the test cases that I would rather not rewrite.